backend/api/server.js

```diff
@@ -45,6 +46,259 @@ server.post('/login', async (req, res, next) => {
     return res.status(401).json("Invalid credentials!");
 });
 
+server.post('/logout', async (req, res, next) => {
+    res.clearCookie('token');
+    return res.status(200).json('User logged out!');
+});
+
+
+/* --- Patient Endpoints --- */
+
+server.get('/patients/:patientId', async (req, res, next) => {
+    const patientId = Number(req.params.patientId);
+    const patient = await prisma.patient.findUnique({
+        where: { id: patientId },
+        include: { user: true }
+    });
+
+    if (!patient) {
+        return res.status(204).json(`No patient with id: ${patientId}`);
+    }
+
+    return res.status(200).json(patient);
+});
+
+server.get('/patients/:patientId/provider', async (req, res, next) => {
+    const patientId = Number(req.params.patientId);
+    const patient = await prisma.patient.findUnique({
+        where: { id: patientId },
+        include: {
+            provider: {
+                include: {
+                    user: true
+                }
+            }
+        }
+    });
+
+    if (!patient) {
+        return res.status(204).json(`No patient with id: ${patientId}`);
+    }
+
+    if (!patient.provider) {
+        return res.status(204).json(`No patient has no provider!`);
+    }
+
+    return res.status(200).json(patient.provider);
+});
+
+server.put('/patients/:patientId/provider', async (req, res, next) => {
+    const patientId = Number(req.params.patientId);
+    const { providerId } = req.body;
+
+    const patient = await prisma.patient.update({
+        where: { id: patientId },
+        include: { user: true },
+        data: { provider_id: providerId }
+    });
+
+    if (!patient) {
+        return res.status(204).json(`No patient with id: ${patientId}`);
+    }
+
+    patient.provider_id = provider_id;
```

Comment:
If we need to manually update it here, does this mean `prisma.patient.update` will return the patient object before it updates the provider_id?